_adr/87/index.adoc

```diff
@@ -1,48 +1,452 @@
+A Prometheus instance is deployed in each of the data plane cluster which uses https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write[remote-write] to push these metrics to Observatorium.
+This acts as the central repository for all metrics, included in the remote-write configuration, from each data plane cluster.
+A list of the metrics included in the remote-write configuration can be found in the https://github.com/bf2fc6cc711aee1a0c2a/observability-resources-mk/blob/main/resources/prometheus/remote-write.yaml[Observability resources] repository.
+
+KAS Fleet Manager acts as a proxy to Observatorium.
+It calls the Observatorium _/query_ and _/query_range_ endpoints to retrieve metrics we consider as 'customer facing'.
+A list of the metrics retrieved by KAS Fleet Manager can be found in the https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/observatorium/api.go[observatorium] client within the project. When new metrics are added to this list, they become available in the KAS Fleet Manager _/metrics/query_ and _/metrics/query_range_ endpoints. In order to make new metrics available to the _/metrics/federate_ endpoint, their metadata must be added to this https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/constants/metrics.go[file] in KAS Fleet Manager.
+
+KAS Fleet Manager controls the labels exposed for each metric returned in all endpoints. 
+Metrics retrieved from Observatorium are also filtered so that end users will only receive metrics for their own Kafka instance.
+Apart from these, no additional aggregation, filtering or modification of the metric is done by KAS Fleet Manager.
 
 ## Proposal
-// What is the decision being proposed
+### Metric and Label Naming
+All metric and label names must adhere to the policy stated in link:../../_ap/16/index.adoc[AP-16: Naming Policy for Customer Facing Metrics].
+
+### Kafka Metric Categories
+The customer facing Kafka metrics can be divided into multiple categories.
+Categorizing these metrics into logical groups can help us formulate the metric name and document it.
+
+|===
+|Category | Description | Sub-component (to be specified in the metric name)
+
+|Cluster
+|General Kafka metrics not specific to any other sub-components.
+|cluster
+
+|Broker
+|Metrics associated to Kafka brokers
+|broker
+
+|Topic
+|Metrics associated to Kafka topics
+|topic (Even though the topic metrics already have a 'topic' label, including this improves the clarity of the metric name)
+|===
+
+Examples of sub-component usage in the metric name:
+
+* For *Cluster* metrics, the metric prefix would be `kas_cluster_`.
+* For *Broker* metrics, the metric prefix would be `kas_broker_`.
+* For *Topic* metrics, the metric prefix would be `kas_topic_`.
+
+### Converted Metrics
+At minimum, each metric should have an `instance_name` label that identifies the Kafka instance if possible.
```

Comment:
+1

Reply:
Should we also expose the instance id?

Reply:
When retrieving metrics from the API, you'd need to specify the Kafka ID. In this case, the instance id should already be known. However, one case where I think this would be useful for is when these metrics are integrated to a metric platform. It would help identify which kafka instance the metric belongs to if a user has federated multiple Kafka metrics to one metric platform. 

@k-wall  Is it possible to expose the instance id as part of the raw kafka metrics?

Reply:
Yes -  we've got several options.  One option is to take the instance id out of the kafka namespace (and removed the kafka- portion).  There's probably a better option to put the id on the pod resources as a label, then rely on the fact that kube labels belome prometheus labels.

Reply:
We use options #1 a lot with the billing metrics: `label_replace` can take the id from the namespace and put it into a new label.

Reply:
I've added this topic to the epic brief that's linked in the description of this epic in JIRA [MGDSTRM-7080](https://issues.redhat.com/browse/MGDSTRM-7080)